src/components/Tooltip/Tooltip.js

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from 'react'
```

Comment:
Do you want to have a go at converting this to hooks? Not a big deal, but might be a good standalone component to start with

Reply:
I rewrote something like below but getting `Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
` error.


```js
const TooltipContainer = (props) => {
  const { text, position, children } = props
  const id = uuid()
  const [show, setShow] = useState(false);
  const [elem, setElem] = useState();

  const _onRef = e => {
    if(e){
      setElem(e)
    }
  }

  useEffect(() => {
    if (show) {
      ReactTooltip.show(findDOMNode(elem))
    } else {
      ReactTooltip.hide(findDOMNode(elem))
    }
  })

  return (
    <>
      {children({
        showTooltip: () => { setShow(true) } ,
        hideTooltip: () => { setShow(false) } ,
        tooltipElement: (
          <span data-tip={text} data-for={id} ref={_onRef} />
        )
      })}
      <DefaultTooltip
        data-testid='tooltip'
        id={id}
        event="dbclick"
        place={position || 'top'}
        effect="solid"
        type="light"
        html={true}
      >
      </DefaultTooltip>
    </>
  )
}

TooltipContainer.propTypes = {
  children: PropTypes.func.isRequired
}
export default TooltipContainer
```